### PR TITLE
added a getFooter function for fetching footer info from contentful

### PR DESCRIPTION
**Pull request by cpualei**
# Pull Request Process

## Copy and Paste / Write ticket description

Footer is not editable/a Contentful component

## Describe the changes you've made to resolve the issue

- Added a getFooter function for fetching footer information from contentful within `app/utils/contentful.ts`.
- Passed footer (getFooter function) prop to Footer component within `app/layout.tsx`.
- Mapped over data from Contentful and removed hard-coded data within `Footer.tsx`.

## Add screenshots of the UI before and after your changes have been made

No changes to the UI

## Code Review Checklist

- [ ] Have the package.json or package-lock.json been changed? If so, does the pull request description say why?
- [ ] Has the UI for mobile and desktop been changed, did you test on IphoneXr and desktop views?
- [x] Does the UI match the requirements in the ticket and designs in figma; does it pass the visual smell test?
- [x] Have TypeScript files been changed, does the TypeScript pass the smell test?
- [ ] If you don't feel super confident in your review, did you assign someone more senior to double check?

### CODE REVIEW

File: app/components/Footer.tsx [hunk 1]
```diff
@@ -9,28 +9,15 @@ import xlogo from "@/public/XLogo.svg";
...
-  text: string;
-  href: string;
-};
-
-type FooterItemsSectionProps = {
-  title: string;
-  id: string;
-  menuItems: FooterItem[];
-};
+import { FooterItemsSectionProps, NavigationBarType } from "../constants/types";
```
Comment (jtorreggiani) [inline app/components/Footer.tsx:20]: You can use the `@/app/constants/` path here.

File: app/components/Footer.tsx [hunk 2]
```diff
@@ -45,39 +32,10 @@ const socialMediaLinks = [
...
-  { text: "Donate", href: "/ways-to-give" },
-  { text: "Events", href: "/events" },
-  { text: "Volunteer", href: "/ways-to-give" },
-];
-
 const styles = {
   footerContainer: "flex justify-evenly bg-[#0076AD] text-white py-8",
   leftContainer: "flex flex-col tracking-wide gap-2 pl-4 lg:pl-4",
-  rightContainer: "hidden lg:flex flex-row gap-x-20",
+  rightContainer: "hidden lg:grid grid-cols-2 md:grid-cols-2 gap-10",
```
Comment (jtorreggiani) [inline app/components/Footer.tsx:38]: @cpualei, the PR description mentions there are no visual changes. What are this changes here for?

**Review by jtorreggiani — APPROVED** [review]
This PR looks good. Just one question about style changes and a recommendation for using the file path alias. Otherwise, I think this is good to merge.

File: app/components/Footer.tsx (hunk 2 above)
Comment (cpualei) [reply app/components/Footer.tsx:38]: Hi @jtorreggiani. That style change was meant to place the right side sections in columns.

This was how it looked before:
Screenshot 2024-06-03 at 3 42 08 PM

And the style change I implemented placed them into two columns like how it looked prior to making the footer an editable Contentful component. Like below:
Screenshot 2024-06-03 at 3 44 06 PM

**Pull request merged**